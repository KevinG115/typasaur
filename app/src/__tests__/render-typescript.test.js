const fs = require("fs");
const path = require("path");
const { inferTypeFromValue } = require("../infer");
const { renderTypescriptFromTree } = require("../renderers/typescript");

describe("renderTypescriptFromTree", () => {
  const options = {
    detectDatesFromIsoStrings: true,
    stringEnumMinUniqueValues: 2,
    stringEnumMaxUniqueValues: 12
  };

  test("renders clean types with field-based names and header", () => {
    const input = {
      email: "a@b.com",
      id: 1,
      isActive: true,
      signupDate: "2025-08-16T12:34:56Z",
      profile: { age: 30, bio: null, socialLinks: ["x", "y"] },
      projects: [{ id: "p1", title: "T", completed: false }],
      roles: ["admin", "editor"]
    };

    const tree = inferTypeFromValue(input, options);
    const ts = renderTypescriptFromTree({
      rootTypeName: "User",
      useInterfaceKeyword: false,
      rootTypeTree: tree
    });

    expect(ts).toMatchSnapshot();
  });

  test("renders complex fixture deterministically", () => {
    const json = JSON.parse(
      fs.readFileSync(path.join(__dirname, "..", "test-data", "complex.json"), "utf8")
    );
    const tree = inferTypeFromValue(json, options);
    const ts = renderTypescriptFromTree({
      rootTypeName: "Model",
      useInterfaceKeyword: true,
      rootTypeTree: tree
    });

    expect(ts).toMatchSnapshot();
  });
});