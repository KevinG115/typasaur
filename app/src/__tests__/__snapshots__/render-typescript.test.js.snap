// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderTypescriptFromTree renders clean types with field-based names and header 1`] = `
"/**
 * Generated by typasaur
 *
 *                    __
 *                   / _)
 *          .-^^^-/ /
 *      __/       /
 *     <__.|_|-|_|
 *
 * JSON to TypeScript Model CLI
 */

export type Profile = {
  age: number;
  bio: any;
  socialLinks: ("x" | "y")[];
};

export type Project = {
  completed: boolean;
  id: string;
  title: string;
};

export type User = {
  email: string;
  id: number;
  isActive: boolean;
  profile: Profile;
  projects: Project[];
  roles: ("admin" | "editor")[];
  signupDate: string | Date;
};
"
`;

exports[`renderTypescriptFromTree renders complex fixture deterministically 1`] = `
"/**
 * Generated by typasaur
 *
 *                    __
 *                   / _)
 *          .-^^^-/ /
 *      __/       /
 *     <__.|_|-|_|
 *
 * JSON to TypeScript Model CLI
 */

export interface Contact = {
  isPrimary: boolean;
  type: string;
  value: string;
};

export interface Social = {
  github: string;
  linkedin: any;
  twitter: string;
};

export interface Profile = {
  bio: any;
  contacts: Contact[];
  dateOfBirth: string | Date;
  firstName: string;
  lastName: string;
  social: Social;
};

export interface Project = {
  contributors: any[];
  id: string;
  status: string;
  tags: ("typescript" | "cli" | "open-source" | "angular" | "monaco" | "frontend")[];
  title: string;
};

export interface Notifications = {
  email: boolean;
  push: any;
  sms: boolean;
};

export interface Settings = {
  language: string;
  notifications: Notifications;
  theme: string;
};

export interface Model = {
  createdAt: string | Date;
  email: string;
  id: number;
  profile: Profile;
  projects: Project[];
  roles: ("admin" | "editor" | "viewer")[];
  settings: Settings;
  updatedAt: any;
  username: string;
};
"
`;
